FIRO_TSEnsembles/src/main/java/hec/JdbcDatabase.java

```diff
@@ -369,111 +417,256 @@ private void InsertEnsemble(int id, int ensemble_timeseries_id, ZonedDateTime is
         ps_insertEnsemble.setBytes(9, byte_value_array);
         ps_insertEnsemble.execute();
     }
+    @Override
+    public List<RecordIdentifier> getTimeSeriesIDs() {
 
-    private int GetMaxID(String tableName) {
-        PreparedStatement p;
+        List<RecordIdentifier> rval = new ArrayList<>();
+        String sql = "select location, parameter_name from " + ensembleTimeSeriesTableName
+                + " order by location,parameter_name";
         try {
-            String sql = "SELECT max(id) max FROM " + tableName;
-            p = _connection.prepareStatement(sql);
+            PreparedStatement stmt = _connection.prepareStatement(sql);
+            ResultSet rs = stmt.executeQuery();
+            // loop through the result set
+            while (rs.next()) {
+                RecordIdentifier tsid = new RecordIdentifier(rs.getString(1), rs.getString(2));
+                rval.add(tsid);
+            }
         } catch (Exception e) {
-            Logger.logError(e.getMessage());
-            return -1;
+            Logger.logError(e);
         }
-        return ScalarQuery(p);
+        return rval;
     }
-
-    private int ScalarQuery(PreparedStatement p) {
-        int rval = 0;
+    @Override
+    public List<ZonedDateTime> getEnsembleIssueDates(RecordIdentifier timeseriesID) {
+        List<ZonedDateTime> rval = new ArrayList<>();
+        PreparedStatement p = null;
         try {
+            String sql = "SELECT issue_datetime from view_ensemble " + " Where location = ?  AND  parameter_name = ?";
+            p = _connection.prepareStatement(sql);
+            p.setString(1, timeseriesID.location);
+            p.setString(2, timeseriesID.parameter);
+
             ResultSet rs = p.executeQuery();
             while (rs.next()) {
-                Object o = rs.getObject(1);
-                if (o == null)
-                    return 0;
-                rval = (int) o;
+                ZonedDateTime t = DateUtility.parseDateTime(rs.getString("issue_datetime"));
+                rval.add(t);
             }
         } catch (Exception e) {
             Logger.logError(e);
         }
         return rval;
     }
+    //endregion
+    //region MetricDatabase
+    @Override
+    public void write(MetricCollectionTimeSeries mts) throws Exception {
+        write(new MetricCollectionTimeSeries[] { mts });
+    }
 
-    private String createTables() throws Exception {
-        runResourceSQLScript("/database.sql");
+    @Override
+    public void write(MetricCollectionTimeSeries[] mtsArray) throws Exception {
+        String compress = "gzip";
+        int mc_id = GetMaxID(metricCollectionTableName);
```

Comment:
If we have a GetNextID() method we could avoid needing ++mc_ts_id and ++mc_id.
GetNextID() could find max and increment.

Reply:
good idea - i think i can do that